build fails due to missing include <weston/compositor.h>

I just tried to build maynard from https://aur.archlinux.org/packages/maynard-git/. The build process was aborted with this error being the culprit:

`shell-helper.c:25:31: fatal error: weston/compositor.h: No such file or directory`

A quick check confirmed that the file didn't exist where I expected it should in `usr/include/weston`, however a couple other files were there. I checked the git repository of weston and discovered that the file compositor.h had been removed between releases 1.11.1 and 1.12.0 (2016-09-20). In fact, the entire tree was restructured. See for yourself:

https://cgit.freedesktop.org/wayland/weston/tree/?id=1.12.0
https://cgit.freedesktop.org/wayland/weston/tree/?id=1.11.1 (`compositor.h` is in `src`)

--- a/configure.ac
+++ b/configure.ac
@@ -34,6 +34,16 @@ PKG_CHECK_MODULES([GTK], [
 	alsa
 ])
 
+PKG_CHECK_MODULES([NEW_WESTON], [
+	weston >= 1.12
+	libweston-1
+	],
+	[AC_DEFINE([HAVE_NEW_WESTON],
+	[1],
+	[Using weston 1.12 or newer])],
+	[AC_MSG_WARN([Not using weston 1.12 or newer])]
+)
+
 GLIB_GSETTINGS
 
 WAYLAND_SCANNER_RULES(['$(top_srcdir)/protocol'])

--- a/shell/shell-helper.c
+++ b/shell/shell-helper.c
@@ -22,7 +22,12 @@
 #include <stdio.h>
 #include <assert.h>
 
+#include "config.h"
+#ifdef HAVE_NEW_WESTON=1
 #include <libweston-1/compositor.h>
+#else
+#include <weston/compositor.h>
+#endif
 
 #include "shell-helper-server-protocol.h"
 
@@ -71,7 +76,11 @@ shell_helper_move_surface(struct wl_client *client,
 static void
 configure_surface(struct weston_surface *es, int32_t sx, int32_t sy)
 {
+	#ifdef HAVE_NEW_WESTON
 	struct weston_view *existing_view = es->committed_private;
+	#else
+	struct weston_view *existing_view = es->configure_private;
+	#endif
 	struct weston_view *new_view;
 
 	new_view = container_of(es->views.next, struct weston_view, surface_link);
@@ -97,12 +106,21 @@ shell_helper_add_surface_to_layer(struct wl_client *client,
 	struct weston_view *new_view, *existing_view, *next;
 	struct wl_layer *layer;
 
+	#ifdef HAVE_NEW_WESTON
 	if (new_surface->committed) {
 		wl_resource_post_error(new_surface_resource,
 				       WL_DISPLAY_ERROR_INVALID_OBJECT,
 				       "surface role already assigned");
 		return;
 	}
+	#else
+	if (new_surface->configure) {
+		wl_resource_post_error(new_surface_resource,
+				       WL_DISPLAY_ERROR_INVALID_OBJECT,
+				       "surface role already assigned");
+		return;
+	}
+	#endif
 
 	existing_view = container_of(existing_surface->views.next,
 				     struct weston_view,
@@ -112,15 +130,24 @@ shell_helper_add_surface_to_layer(struct wl_client *client,
 		weston_view_destroy(new_view);
 	new_view = weston_view_create(new_surface);
 
+	#ifdef HAVE_NEW_WESTON
 	new_surface->committed = configure_surface;
 	new_surface->committed_private = existing_view;
+	#else
+	new_surface->configure = configure_surface;
+	new_surface->configure_private = existing_view;
+	#endif
 	new_surface->output = existing_view->output;
 }
 
 static void
 configure_panel(struct weston_surface *es, int32_t sx, int32_t sy)
 {
+	#ifdef HAVE_NEW_WESTON
 	struct shell_helper *helper = es->committed_private;
+	#else
+	struct shell_helper *helper = es->configure_private;
+	#endif
 	struct weston_view *view;
 
 	view = container_of(es->views.next, struct weston_view, surface_link);
@@ -147,7 +174,11 @@ shell_helper_set_panel(struct wl_client *client,
 	 * it hasn't yet been defined because the original surface configure
 	 * function hasn't yet been called. if we call it here we will have
 	 * access to the layer. */
+	#ifdef HAVE_NEW_WESTON
 	surface->committed(surface, 0, 0);
+	#else
+	surface->configure(surface, 0, 0);
+	#endif
 
 	helper->panel_layer = container_of(view->layer_link.link.next,
 					   struct weston_layer,
@@ -155,8 +186,13 @@ shell_helper_set_panel(struct wl_client *client,
 
 	/* set new configure functions that only ensure the surface is in the
 	 * correct layer. */
+	#ifdef HAVE_NEW_WESTON
 	surface->committed = configure_panel;
 	surface->committed_private = helper;
+	#else
+	surface->configure = configure_panel;
+	surface->configure_private = helper;
+	#endif
 }
 
 enum SlideState {

